packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue

```diff
@@ -1,66 +1,12 @@
 <template>
   <div class="link-details oc-flex oc-flex-between oc-flex-middle">
     <div v-if="isModifiable">
-      <oc-button
-        :id="`edit-public-link-role-dropdown-toggle-${link.id}`"
-        appearance="raw"
-        class="edit-public-link-role-dropdown-toggle oc-text-left"
-        gap-size="none"
-      >
-        <span
-          class="link-current-role"
-          v-text="currentLinkRoleLabel || $gettext('Select a role')"
-        />
-        <oc-icon name="arrow-down-s" />
-      </oc-button>
-      <oc-drop
-        ref="editPublicLinkRoleDropdown"
-        class="edit-public-link-role-dropdown"
-        :drop-id="`edit-public-link-role-dropdown`"
-        :toggle="`#edit-public-link-role-dropdown-toggle-${link.id}`"
-        padding-size="small"
-        offset="0"
-        mode="click"
-      >
-        <oc-list class="role-dropdown-list">
-          <li v-for="roleOption in availableRoleOptions" :key="`role-dropdown-${roleOption.key}`">
-            <oc-button
-              :id="`files-role-${roleOption.name}`"
-              :class="{
-                selected: isSelectedRole(roleOption),
-                'oc-background-primary-gradient': isSelectedRole(roleOption)
-              }"
-              :appearance="isSelectedRole(roleOption) ? 'raw-inverse' : 'raw'"
-              :variation="isSelectedRole(roleOption) ? 'primary' : 'passive'"
-              justify-content="space-between"
-              class="oc-p-s"
-              @click="
-                updateLink({
-                  link: {
-                    ...link,
-                    permissions: roleOption.bitmask(false)
-                  },
-                  dropRef: $refs.editPublicLinkRoleDropdown
-                })
-              "
-            >
-              <span class="oc-flex oc-flex-middle">
-                <oc-icon :name="roleOption.icon" class="oc-pl-s oc-pr-m" variation="inherit" />
-                <span>
-                  <span
-                    class="oc-text-bold oc-display-block oc-width-1-1"
-                    v-text="$gettext(roleOption.label)"
-                  />
-                  <span class="oc-text-small">{{ $gettext(roleOption.description(false)) }}</span>
-                </span>
-              </span>
-              <span class="oc-flex">
-                <oc-icon v-if="isSelectedRole(roleOption)" name="check" variation="inherit" />
-              </span>
-            </oc-button>
-          </li>
-        </oc-list>
-      </oc-drop>
+      <link-role-dropdown
+        v-model="currentLinkRole"
+        :available-role-options="availableRoleOptions"
+        drop-offset="0"
+        @update:model-value="updateSelectedRole"
```

Comment:
I know we do this sometimes, but I think it's a bit ugly to use v-model and update handlers together as you never know which one fires first.. so maybe set the new value in updateSelectedRole and use `:model-value`?

Reply:
Yep I agree, that's cleaner 👍